test/claims-status/utils/appealsV2Actions.unit.spec.js

```diff
@@ -57,15 +63,26 @@ describe('getAppealsV2', () => {
       }).then(done, done);
   });
 
-  it('dispatches FETCH_APPEALS_UNAVAILABLE', (done) => {
-    global.fetch.returns(Promise.reject());
-    const thunk = getAppealsV2();
-    const dispatch = sinon.spy();
-    thunk(dispatch)
-      .then(() => {
-        const action = dispatch.secondCall.args[0];
-        expect(action.type).to.equal(SET_APPEALS_UNAVAILABLE);
-      }).then(done, done);
-  });
+  const appealsErrors = {
+    403: USER_FORBIDDEN,
+    404: RECORD_NOT_FOUND,
+    422: VALIDATION_ERROR,
+    502: BACKEND_SERVICE_ERROR,
+    504: APPEALS_FETCH_ERROR // works for any unspecified error code
+  };
 
+  Object.keys(appealsErrors).forEach((code) => {
+    it(`Dispatches ${appealsErrors[code]} when GET fails with ${code}`, (done) => {
+      global.fetch.returns(Promise.reject({
+        errors: [{ status: `${code}` }],
+      }));
+      const thunk = getAppealsV2();
+      const dispatch = sinon.spy();
+      thunk(dispatch)
+        .then(() => {
+          const action = dispatch.secondCall.args[0];
+          expect(action.type).to.equal(appealsErrors[code]);
+        }).then(done, done);
```

Comment:
Thinking this through...

If the `expect` fails, will it be caught by the `.then(done, done)`? Or would it only be caught if there were a `.catch()`? I always forget when the error callback on the `.then()` is called.

Anyhow, if the `.then()` error callback is called, it'll call `done()` with no argument, which will tell the test it succeeded when it actually failed, so maybe just verify that.

Reply:
Confirmed, works 👍